core/jobs/update_integration_status.py

```diff
@@ -0,0 +1,76 @@
+import datetime
+
+from sqlalchemy import func
+
+from core.model import ExternalIntegration, ExternalIntegrationError
+from core.scripts import Script
+from core.util.datetime_helpers import utc_now
+
+
+class UpdateIntegrationStatusScript(Script):
+    """Script to update the external integration status fields
+    based on any errors logged in the db
+    """
+
+    # 10 errors in the last 30 minutes
+    ERROR_WINDOW_DELTA = datetime.timedelta(minutes=30)
+    ERROR_WINDOW_COUNT = 10
```

Comment:
I forgot to draw attention to these numbers, they were chosen because they are numbers, and for no other reason.
Should they be changed to something with more rationale behind them?
@dbernstein @jonathangreen

Reply:
@RishiDiwanTT :  I like numbers.  Numbers are good.  ;) It seems like as good a guess as any.  I think 10 errors in 1/2  hour is a good place to start.